hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -30,33 +31,42 @@ import Hydra.ServerOutput (ServerOutput (..))
 import Hydra.Snapshot (Snapshot (..), SnapshotNumber)
 
 data Event tx
-  = ClientEvent (ClientInput tx)
-  | NetworkEvent (Message tx)
-  | OnChainEvent (OnChainTx tx)
+  = ClientEvent {clientInput :: ClientInput tx}
+  | NetworkEvent {message :: Message tx}
+  | OnChainEvent {onChainTx :: OnChainTx tx}
   | ShouldPostFanout
   | DoSnapshot
   deriving stock (Eq, Show, Generic)
   deriving anyclass (ToJSON, FromJSON)
 
+instance (Arbitrary tx, Arbitrary (UTxO tx)) => Arbitrary (Event tx) where
+  arbitrary = genericArbitrary
+
 data Effect tx
-  = ClientEffect (ServerOutput tx)
-  | NetworkEffect (Message tx)
-  | OnChainEffect (OnChainTx tx)
-  | Delay DiffTime (Event tx)
+  = ClientEffect {serverOutput :: ServerOutput tx}
+  | NetworkEffect {message :: Message tx}
+  | OnChainEffect {onChainTx :: OnChainTx tx}
+  | Delay {delay :: DiffTime, event :: Event tx}
   deriving stock (Generic)
 
+instance (Arbitrary tx, Arbitrary (UTxO tx)) => Arbitrary (Effect tx) where
+  arbitrary = genericArbitrary
+
 deriving instance Tx tx => Eq (Effect tx)
 deriving instance Tx tx => Show (Effect tx)
 deriving instance Tx tx => ToJSON (Effect tx)
 deriving instance Tx tx => FromJSON (Effect tx)
 
 data HeadState tx
   = ReadyState
-  | InitialState HeadParameters PendingCommits (Committed tx)
-  | OpenState HeadParameters (CoordinatedHeadState tx)
-  | ClosedState HeadParameters (UTxO tx)
+  | InitialState {parameters :: HeadParameters, pendingCommits :: PendingCommits, committed :: Committed tx}
+  | OpenState {parameters :: HeadParameters, coordinatedHeadState :: CoordinatedHeadState tx}
+  | ClosedState {parameters :: HeadParameters, utxos :: UTxO tx}
```

Comment:
This is interesting: Is `UTxO` singular or plural now? i.e. do we refer to many unspent transaction outputs or only a single one here? Not sure what to change, the field name or the type name though :sweat_smile:

Reply:
This is a quick fix to resolve a compiler ambiguity. I _thought_ UTxO meant _Unspent Transaction Outputs_ so it's already a plural, but actually I am wrong: https://en.wikipedia.org/wiki/Unspent_transaction_output Also note that Wikipedia capitalise all the letters... 🤷